### PR TITLE
feat: develop Additional Inquiries Form for User Questions on the FAQ…

### LINKED ISSUE #148: [Feature]: Develop Additional Questions Form for User Inquiries on the FAQ page
### Description

Develop an Additional Questions form where users can submit specific inquiries or questions that are not addressed in the FAQs. Ensure the form includes fields for user's name, email address, phone number (optional), and message. Implement form validation, integrate with backend API endpoints for submission handling, and provide feedback messages upon submission.

### Acceptance Criteria

- [ ] Develop a form with fields for user's name, email address and message.
- [ ] Implement form validation to ensure required fields are filled and valid email format is entered.
- [ ] Integrate the form with backend API endpoints to handle form submissions
- [ ] Display success or error messages based on API response upon form submission.
- [ ] Ensure the form matches the provided design specifications and is responsive
- [ ] Write unit tests to verify form validation, API integration, and responsiveness.

**Deadline: Friday 10pm**

### Links

https://www.figma.com/design/VEItfX6St5NSAqqNHImcxD/HNG-Boilerplate-Designs?node-id=432-7537&m=dev
image

### PR DESCRIPTION
**Pull request by Ralpmike**
… page

# Description
Develop an Additional Questions form where users can submit specific inquiries or questions that are not addressed in the FAQs. Ensure the form includes fields for user's name, email address, phone number (optional), and message. Implement form validation, integrate with backend API endpoints for submission handling, and provide feedback messages upon submission.

**Closes #148**

# Changes proposed

## What were you told to do?

- Develop a form with fields for user's name, email address and message.
- Implement form validation to ensure required fields are filled and valid email format is entered.
- Integrate the form with backend API endpoints to handle form submissions
- Display success or error messages based on API response upon form submission.
- Ensure the form matches the provided design specifications and is responsive
- Write unit tests to verify form validation, API integration, and responsiveness

.

## What did you do?

- Created a new AdditionalInquiriesForm component with fields for the user's name, email address, and message.
- Implemented form validation using [Zod](https://github.com/colinhacks/zod) to ensure all required fields are filled and the email format is valid.
- Integrated the form with backend API endpoints for handling form submissions.
- Added feedback messages to display success or error responses based on the API response.
- Ensured the form is responsive and matches the provided design specifications.
- Wrote unit tests to verify the form validation, API integration, and responsiveness.

# Check List (Check all the applicable boxes)

🚨Please review the [contribution guideline](CONTRIBUTING.md) for this repository.

- [x] My code follows the code style of this project.
- [x] This PR does not contain plagiarized content.
- [x] The title and description of the PR is clear and explains the approach.
- [x] I am making a pull request against the **dev branch** (left side).
- [x] My commit messages styles matches our requested structure.
- [x] My code additions will fail neither code linting checks nor unit test.
- [x] I am only making changes to files I was requested to.

# Screenshots/Videos

image

image

### CODE REVIEW

**Comment by Ralpmike** [comment]
Hey @Samadeen ,

The PR for the Additional Inquiries form is ready for review. 
Could you take a look and let me know if there are any changes needed? Thanks in advance!

Eslint test:
image

Vitest:
image

**Review by SirhmVFX — APPROVED** [review]

File: src/app/additionalinquiriesform/AdditionalInquiriesForm.tsx [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+
+  const [successMessage, setSuccessMessage] = useState("");
+  const [errorMessage, setErrorMessage] = useState("");
+
+  async function onSubmit(values: z.infer<typeof FormSchema>) {
+    // eslint-disable-next-line no-console
+    console.log(values);
+
+    try {
+      const response = await fetch("/api/contact", {
```
Comment (oreoluwa212) [inline src/app/additionalinquiriesform/AdditionalInquiriesForm.tsx]: who gave you this API to use?

static page

**Review by oreoluwa212 — CHANGES_REQUESTED** [review]
Almost good but we dont want fetch for now

File: src/app/additionalinquiriesform/AdditionalInquiriesForm.tsx (hunk 1 above)
Comment (Ralpmike) [reply src/app/additionalinquiriesform/AdditionalInquiriesForm.tsx]: It's just a dummy text

**Comment by Ralpmike** [comment]
> Almost good but we dont want fetch for now

what then should I use in that regards?

**Comment by Ralpmike** [comment]
@oreoluwa212 
I have replaced the fetch API call in the `AdditionalInquiriesForm` component with dummy data as instructed. Please review for a possible merge.
image

image

**Comment by Ralpmike** [comment]
@JulianaSau,
Could you please review my PR.

**Review by JulianaSau — APPROVED** [review]

**Comment by JulianaSau** [comment]
update your branch

**Comment by Ralpmike** [comment]
@JulianaSau,
Done updating my branch

**Comment by Ralpmike** [comment]
@JulianaSau, @oreoluwa212 ,
could you please merge my branch. thanks

**Pull request merged**